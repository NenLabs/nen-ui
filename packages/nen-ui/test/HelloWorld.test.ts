import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import HelloWorld from '../src/components/HelloWorld.vue'

describe('HelloWorld.vue', () => {
  it('should render', () => {
    const wrapper = mount(HelloWorld, { props: { msg: 'I am a message' } })
    expect(wrapper.text()).toContain('I am a message')
  })

  it('should be interactive', async () => {
    const wrapper = mount(HelloWorld, { props: { msg: 'I am a message' } })
    expect(wrapper.text()).toContain('I am a message')

    expect(wrapper.find('.increment').exists()).toBe(true)

    expect(wrapper.find('.dec').exists()).toBe(false)

    await wrapper.get('.increment').trigger('click')

    expect(wrapper.text()).toContain('1')
  })
})