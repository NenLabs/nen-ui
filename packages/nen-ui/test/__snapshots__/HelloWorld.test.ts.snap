// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HelloWorld.vue > should render 1`] = `
"<h1 data-v-e17ea971=\\"\\">I am a message</h1>
<div data-v-e17ea971=\\"\\" class=\\"card\\"><button data-v-e17ea971=\\"\\" type=\\"button\\" class=\\"increment\\">count is 0</button>
  <p data-v-e17ea971=\\"\\"> Edit <code data-v-e17ea971=\\"\\">components/HelloWorld.vue</code> to test HMR </p>
</div>
<p data-v-e17ea971=\\"\\"> Check out <a data-v-e17ea971=\\"\\" href=\\"https://vuejs.org/guide/quick-start.html#local\\" target=\\"_blank\\">create-vue</a>, the official Vue + Vite starter </p>
<p data-v-e17ea971=\\"\\"> Install <a data-v-e17ea971=\\"\\" href=\\"https://github.com/vuejs/language-tools\\" target=\\"_blank\\">Volar</a> in your IDE for a better DX </p>
<p data-v-e17ea971=\\"\\" class=\\"read-the-docs\\">Click on the Vite and Vue logos to learn more</p>"
`;
